### PR TITLE
Automated cherry pick of #28770 #30828

### PR DESCRIPTION
**Pull request by bprashanth**
Cherry pick of #28770 #30828 on release-1.3.

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/30832)

### CODE REVIEW

**Comment by bprashanth** [comment]
Looks to me like the bottom commit/first pr was just a refactor that moved a bunch of methods into utility functions, so I'm picking them both to avoid hand rebasing. @gmarek sanity check would help.

File: pkg/controller/node/deletion_utils.go [hunk 1]
```diff
@@ -224,6 +232,23 @@ func markAllPodsNotReady(kubeClient clientset.Interface, nodeName string) error
...
 }
 
+// nodeRunningOutdatedKubelet returns true if the kubeletVersion reported
+// in the nodeInfo of the given node is "outdated", meaning < 1.2.0.
+// Older versions were inflexible and modifying pod.Status directly through
+// the apiserver would result in unexpected outcomes.
+func nodeRunningOutdatedKubelet(node *api.Node) bool {
+	v, err := version.Parse(node.Status.NodeInfo.KubeletVersion)
+	if err != nil {
+		glog.Errorf("couldn't parse version %q of node %v", node.Status.NodeInfo.KubeletVersion, err)
```
Comment (gmarek) [inline pkg/controller/node/deletion_utils.go]: add: "assuming outdated Kubelet version" to the error string.

**Comment by gmarek** [comment]
LGTM

File: pkg/controller/node/deletion_utils.go (hunk 1 above)
Comment (bprashanth) [inline pkg/controller/node/deletion_utils.go]: I'll do this in HEAD with a follow up, I want this pr to match the parent

**Comment by bprashanth** [comment]
Re-applying lgtm, I'd dropped an import rebasing

**Comment by bprashanth** [comment]
fixed boilerplate, inheriting lgtm

**Pull request merged**